desdeo_emo/selection/APD_Select_constraints.py

```diff
@@ -83,6 +83,28 @@ def do(self, pop: Population, vectors: ReferenceVectors) -> List[int]:
             sub_population_index = np.atleast_1d(
                 np.squeeze(np.where(assigned_vectors == i))
             )
+
+            # Constraint check
+            if len(sub_population_index) > 1 and pop.constraint_violation is not None:
+                constraint_following_index = []
+                violation_values = []
+                for individual in sub_population_index:
+                    violation_values.append(pop.constraint_violation[individual])
+
+                    if((pop.constraint_violation[individual] > 0).all()):
+                        constraint_following_index.append(individual)
+
+                violation_values = np.asarray(violation_values)
+                violation_values = violation_values.sum(0)
+
+                # Case when entire subpopulation is infeasible
+                if len(constraint_following_index) == 0:
+                    sub_population_index = sub_population_index[np.where(violation_values == violation_values.max())]
+                # Case when only some are infeasible
+                else:
+                    sub_population_index = constraint_following_index
+            sub_population_index = np.asarray(sub_population_index)
+
             sub_population_fitness = translated_fitness[sub_population_index]
             if len(sub_population_fitness > 0):
```

Comment:
Make a case for sub_population_fitness == 1, and just choose that solution for the next generation, without doing all this unnecessary computation. In the best case scenario, this is inefficient. In the worst case scenario, this will interfere with the "all infeasible solutions" case